model/project_ref.go

```diff
@@ -1921,6 +1923,15 @@ func FindProjectRefs(key string, limit int, sortDir int) ([]ProjectRef, error) {
 	return projectRefs, err
 }
 
+// CanEnableRepotracker checks if the project can enable the repotracker
+// with the current state of their project settings.
+func (p *ProjectRef) ValidateModifiedRepotracker() error {
```

Comment:
in both places we call this function, we also check if the repotracker is enabled. It'd make sense to just check that in here.